File: string/redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.go
package redistributecharacterstomakeallstringsequal

func makeEqual(words []string) bool {
	var count [26]int

	for _, word := range words {
		for _, c := range word {
			count[c-'a']++
		}
	}

	for _, c := range count {
		if c%len(words) != 0 {
			return false
		}
	}

	return true
}

File: string/redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.ts
function makeEqual(words: string[]): boolean {
    const count = new Array<number>(26).fill(0);

    for (let word of words) {
        for (let char of word) {
            const index = char.charCodeAt(0) - "a".charCodeAt(0);
            count[index]++;
        }
    }

    for (let el of count) {
        if ((el !== 0) && (el % words.length !== 0)) {
            return false;
        }
    }

    return true;
};
